agent_id/include/essentials/AgentID.h

```diff
@@ -60,10 +61,23 @@ class AgentID
     static const uint8_t UUID_TYPE = 2;
 
   private:
+    template <class Prototype>
+    void setID(Prototype& idPrototype);
+
     std::vector<uint8_t> _id;
     uint8_t _type;
 };
 
+template <class Prototype>
+void AgentID::setID(Prototype& prototypeID)
+{
+    // little-endian encoding
+    // TODO: replace with memcpy or std copy
+    for (int i = 0; i < static_cast<int>(sizeof(Prototype)); i++) {
```

Comment:
operator uint64_t()  in Agent.cpp has an example on how to handle both little and big endian

Reply:
I wouldn't do anything else here in case of a big-endian architecture. Should we remove the comment?